### DISCUSSION TITLE
Separate task and memory leak

### DISCUSSION THREAD

**Discussion by Carlito06150** [pschatzmann/arduino-audio-tools#1755] [Q&A] [answered]
Hello,

in my setup I call this code:
```
SD.begin(chipSelect);
audioFile = SD.open("/start_song.mp3");
decoder.begin();
copier.begin(decoder, audioFile);      
task.begin([](){copier.copy();});
```

and when I need to play a file, I call this code:
```
void lireSon(const char *name){
    task.suspend();
    delay(1);
    audioFile.close();
    delay(1);
    audioFile = SD.open(name);
    decoder.begin();  
    copier.begin(decoder, audioFile);
    task.resume();
}
```

but each time I call the function, I loose a bit of memory... would you see somthing to do to avoid that?

**Comment by pschatzmann** [comment, marked as answer]
Try to find out where this is happening:
My guess would be that it is caused by your  decoder.begin(): Just call this once if it is a mp3 or end it before calling begin again when it is a wav

**Reply by Carlito06150** [reply]
oh you may be right, I gonna check, tanks a lot
